### ISSUE TITLE
dotnet tools do not support differing entitlements than the dotnet muxer on Mac

### ISSUE THREAD

**Issue by agocke** [dotnet/sdk#23640]
See https://github.com/dotnet/runtime/issues/63952 for context. In this case, a dotnet tool requires more entitlements than `dotnet`, so it can't be executed using the `dotnet` muxer. If dotnet tools always use an apphost, then the host should be ad hoc signed, which would allow all requirements.

**Comment by ClearScriptLib** [comment]
This issue affects `dotnet test` as well.

**Comment by marcpopMSFT** [comment]
@agocke Do we have to increase the entitlements for the host created during tool install? Is there risk in setting more entitlements for dotnet?

**Comment by agocke** [comment]
> Is there risk in setting more entitlements for dotnet?

Yeah, those are minimum install requirements for `dotnet`, so we want that list to be as small as possible. For the dotnet tool, the host that sometimes gets created is ad hoc signed, so it automatically has all entitlements.

**Comment by baronfel** [comment]
I suspect with the new possibility of RID-specific tools, a tool author would be able to ensure that their macos variants have all of the entitlements they require?
